test/e2e/storage/drivers/csi.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * This file defines various csi volume test drivers for TestSuites.
+ *
+ * There are two ways, how to prepare test drivers:
+ * 1) With containerized server (NFS, Ceph, Gluster, iSCSI, ...)
+ * It creates a server pod which defines one volume for the tests.
+ * These tests work only when privileged containers are allowed, exporting
+ * various filesystems (NFS, GlusterFS, ...) usually needs some mounting or
+ * other privileged magic in the server pod.
+ *
+ * Note that the server containers are for testing purposes only and should not
+ * be used in production.
+ *
+ * 2) With server or cloud provider outside of Kubernetes (Cinder, GCE, AWS, Azure, ...)
+ * Appropriate server or cloud provider must exist somewhere outside
+ * the tested Kubernetes cluster. CreateVolume will create a new volume to be
+ * used in the TestSuites for inlineVolume or DynamicPV tests.
+ */
+
+package drivers
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+const (
+	csiExternalProvisionerClusterRoleName string = "system:csi-external-provisioner"
+	csiExternalAttacherClusterRoleName    string = "system:csi-external-attacher"
```

Comment:
These roles are deprecated. Don't use them. Instead each driver deployment must create its own roles.

PR #69868 makes that change. Let's decide about that first, then figure out how to proceed with this PR.

Reply:
@pohly 

Thank you for your comment.
This PR is focusing on sharing the test with in-tree drivers and csi drivers, not focusing on how csi driver will prepare test, like creating sa, clusterrole, ds, and so on. 

It seems an improvement for specific to how to prepare test for csi-hostpath driver. So, I think that we can improve it separately with this PR.

Reply:
@pohly 

This PR also aim to split driver logics and test logics, which has already been done in in-tree driver.
RBAC related change would be only in driver logics, so once this PR is merged, any changes for test logics will be able to merge without conflict to driver logic changes including RBAC related changes. (So, I prioritize this PR over my csi rbd driver PR.)

I don't mean to say that this PR should be prioritized over RBAC PR, and I will rebase this PR if RBAC PR is merged before this PR.